src/content/docs/en/reference/image-service-reference.mdx

```diff
@@ -0,0 +1,181 @@
+---
+title: Image Service API
+i18nReady: false
+---
+
+Astro [assets](/en/guides/assets/) were designed to make it easy for any image optimization service to build a service on top of Astro.
+
+:::caution
+The Image Services API is an experimental Astro feature introduced in v2.1. This API is subject to change before it is marked as stable.
+:::
+
+## What is an Image Service?
+
+Astro provides two types of image services: Local and External.
+
+- **Local services** handle image transformations directly at build for static sites, or at runtime both in development mode and SSR. These are often wrappers around libraries like Sharp, ImageMagick, or Squoosh. In dev mode and in SSR, local services use an API endpoint to do the transformation.
+- **External services** point to URLs and can add support for services such as Cloudinary, Vercel, or any [RIAPI](https://github.com/riapi/riapi)-compliant server.
+
+## Building using the Image Services API
+
+Services definitions take the shape of an exported default object with various required methods ("hooks") to create the required properties for your image. 
+
+Local services have a `transform()` method available to perform transformations on your image. External services can only return the final URL and image attributes.
+
+### Local Service
+
+To create your own local service, you can choose to use the built-in endpoint (`/_image`) and pass your own `transform` function, or you can additionally create your own endpoint that calls the service's functions.
+
+```js
+import type { LocalImageService } from "astro";
+const service: LocalImageService = {
+  getURL(options: ImageTransform) {
+    const href = typeof options.src === "string" ? options.src : options.src.src;
+    return `/my_super_endpoint_that_transforms_images?href={href}&w=${options.width}`;
+  },
+  parseURL(url: URL) {
+    return {
+      src: url.searchParams.get("href"),
+      width: url.searchParams.get("w"),
+    };
+  },
+  transform(buffer: Buffer, options: { src: string, [key: string]: any }): { data: Buffer, format: OutputFormat } {
+    const { buffer } = mySuperLibraryThatEncodesImages(options);
+    return {
+      data: buffer,
+      format: options.format,
+    };
+  },
+};
+export default service;
+```
+
+#### How this Works
+
+At build time for static sites, both `<Image/>` and `getImage(options)` call the `transform()` function. They pass options either through component attributes or an `options` argument, respectively. The transformed images will be built to a `dist/_astro` folder.
```

Comment:
This is also the case for pre-rendered routes, if you want to add an additional precision.